plugin/trino-sybase/src/main/java/io/trino/plugin/sybase/SybaseClient.java

```diff
@@ -0,0 +1,214 @@
+
+import com.google.common.base.Joiner;
+import com.google.common.collect.ImmutableSet;
+import io.trino.plugin.jdbc.BaseJdbcClient;
+import io.trino.plugin.jdbc.BaseJdbcConfig;
+import io.trino.plugin.jdbc.ColumnMapping;
+import io.trino.plugin.jdbc.ConnectionFactory;
+import io.trino.plugin.jdbc.JdbcColumnHandle;
+import io.trino.plugin.jdbc.JdbcTableHandle;
+import io.trino.plugin.jdbc.JdbcTypeHandle;
+import io.trino.plugin.jdbc.WriteMapping;
+import io.trino.plugin.jdbc.mapping.IdentifierMapping;
+import io.trino.spi.TrinoException;
+import io.trino.spi.connector.ConnectorSession;
+import io.trino.spi.connector.SchemaTableName;
+import io.trino.spi.type.CharType;
+import io.trino.spi.type.DecimalType;
+import io.trino.spi.type.Type;
+import io.trino.spi.type.VarcharType;
+
+import javax.inject.Inject;
+
+import java.sql.Connection;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Types;
+import java.util.Collection;
+import java.util.List;
+import java.util.Optional;
+import java.util.function.BiFunction;
+
+import static io.trino.plugin.jdbc.JdbcErrorCode.JDBC_ERROR;
+import static io.trino.plugin.jdbc.StandardColumnMappings.booleanWriteFunction;
+import static io.trino.plugin.jdbc.StandardColumnMappings.charWriteFunction;
+import static io.trino.plugin.jdbc.StandardColumnMappings.varcharWriteFunction;
+import static io.trino.spi.StandardErrorCode.NOT_SUPPORTED;
+import static io.trino.spi.type.BooleanType.BOOLEAN;
+import static java.lang.String.format;
+import static java.util.Objects.requireNonNull;
+import static java.util.stream.Collectors.joining;
+
+public class SybaseClient
+        extends BaseJdbcClient
+{
+    // SqlServer supports 2100 parameters in prepared statement, let's create a space for about 4 big IN predicates
+    public static final int SYBASE_MAX_LIST_EXPRESSIONS = 500;
+
+    private static final Joiner DOT_JOINER = Joiner.on(".");
+
+    @Override
+    public void createSchema(ConnectorSession session, String schemaName)
+    {
+        // TODO: Add try-catch for error message parsing (test cases patch)
+        super.createSchema(session, schemaName);
+    }
+
+    @Inject
+    public SybaseClient(BaseJdbcConfig config, ConnectionFactory connectionFactory, IdentifierMapping identifierMapping)
+    {
+        super(config, "\"", connectionFactory, identifierMapping);
+
+        requireNonNull(config, "sybaseConfig is null");
+    }
+
+    @Override
+    protected void renameTable(ConnectorSession session, String catalogName, String schemaName, String tableName, SchemaTableName newTable)
+    {
+        if (!schemaName.equals(newTable.getSchemaName())) {
+            throw new TrinoException(NOT_SUPPORTED, "This connector does not support renaming tables across schemas");
+        }
+
+        String sql = format(
+                "sp_rename %s, %s",
+                singleQuote(catalogName, schemaName, tableName),
+                singleQuote(newTable.getTableName()));
```

Comment:
As far as I checked this rename logic locally, this method throws an error. Perhaps, Sybase doesn't allow passing catalog nor schema names in this procedure?
http://infocenter.sybase.com/help/index.jsp?topic=/com.sybase.infocenter.dc36273.1572/html/sprocs/X37949.htm

Reply:
let me check on this..